test/specs/parsing/axes/AncestorAxis.tests.ts

```diff
@@ -28,6 +30,26 @@ describe('ancestor', () => {
 		], documentNode);
 		chai.assert.deepEqual(evaluateXPathToStrings('ancestor-or-self::*/position()', documentNode.documentElement.firstChild), ['1', '2']);
 	});
+
+	it('passes buckets for ancestor', () => {
+		jsonMlMapper.parse([
+			'parentElement',
+			['childElement']
+		], documentNode);
+
+		const parentNode = documentNode.firstChild;
+		const childNode = documentNode.firstChild.firstChild;
+
+		const testDomFacade: IDomFacade = {
+			getParentNode: (node, bucket) => {
+				chai.assert.notEqual(bucket, null, 'There must be a bucket passed!');
+				return getBucketsForNode(node.parentNode).includes(bucket) ? node.parentNode : null;
+			}
+		} as any;
+
+		const results = evaluateXPathToNodes('ancestor::parentElement', childNode, testDomFacade);
```

Comment:
When we do the check I described in the comment above, there will be no need left to check the result of xpath.
So this would than just be `evaluateXPathToNodes('ancestor::parentElement', childNode, testDomFacade);`.